core/src/accounts_hash_verifier.rs

```diff
@@ -155,19 +179,46 @@ impl AccountsHashVerifier {
             }
         }
 
-        if let Some(snapshot_config) = snapshot_config {
-            if snapshot_package.block_height % snapshot_config.full_snapshot_archive_interval_slots
-                == 0
-            {
-                if let Some(pending_snapshot_package) = pending_snapshot_package {
-                    *pending_snapshot_package.lock().unwrap() = Some(snapshot_package);
-                }
-            }
-        }
-
         cluster_info.push_accounts_hashes(hashes.clone());
     }
 
+    fn submit_for_packaging(
+        accounts_package: AccountsPackage,
+        pending_snapshot_package: Option<&PendingSnapshotPackage>,
+        snapshot_config: Option<&SnapshotConfig>,
+    ) {
+        if accounts_package.snapshot_type.is_none()
+            || pending_snapshot_package.is_none()
+            || snapshot_config.is_none()
+        {
+            return;
+        };
+
+        let snapshot_package = SnapshotPackage::from(accounts_package);
+        let pending_snapshot_package = pending_snapshot_package.unwrap();
+        let _snapshot_config = snapshot_config.unwrap();
+
+        // If the snapshot package is an Incremental Snapshot, do not submit it if there's already
+        // a pending Full Snapshot.
+        let can_submit = match snapshot_package.snapshot_type {
+            SnapshotType::FullSnapshot => true,
+            SnapshotType::IncrementalSnapshot(_) => pending_snapshot_package
+                .lock()
+                .unwrap()
+                .as_ref()
+                .map_or(true, |snapshot_package| {
+                    matches!(
+                        snapshot_package.snapshot_type,
```

Comment:
a simple snapshot_package.snapshot_type == SnapshotType::IncrementalSnapshot does not work?

Reply:
Yah, since `SnapshotType::IncrementalSnapshot` is a tuple struct with a base slot field, I got a compile error for not accounting for the field. The `matches!` macro is basically just a `switch` under the hood.

Reply:
Got it -- saw you have changed the enum def -- maybe we should define a utility function check is_incremental?

Reply:
I like it. Done!